File: 13_roman_to_integer.cxx
class Solution {
public:
    int romanToInt(string s) {
        std::unordered_map<char,int> symval;
        symval['I'] = 1;
        symval['V'] = 5;
        symval['X'] = 10;
        symval['L'] = 50;
        symval['C'] = 100;
        symval['D'] = 500;
        symval['M'] = 1000;
        
        int out=0;
        for(auto i=s.begin(); i<s.end(); ++i) {
            if(symval[*i]<symval[*(i+1)]) {
                out+= symval[*(i+1)]-symval[*i];
                ++i;
            } else {
                out += symval[*i];
            }
        }
        return out;
    }
};

File: 13_roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        symval = {  'I':1,
                    'V':5,
                    'X':10,
                    'L':50,
                    'C':100,
                    'D':500,
                    'M':1000} 
        out=0
        previous=0
        for i in s[::-1]:
            if symval[i] >= previous:
                out += symval[i]
            else:
                out -= symval[i]
            previous = symval[i]
        return out
